pulp_2to3_migration/app/pre_migration.py

```diff
@@ -296,20 +304,62 @@ def pre_migrate_all_without_content(plan):
             importer_types = list(plugin_plan.migrator.importer_migrators.keys())
+            distributor_migrators = plugin_plan.migrator.distributor_migrators
+            distributor_types = list(distributor_migrators.keys())
+
+            # figure out which repos/importers/distributors have been updated since the last run
+            epoch = datetime.utcfromtimestamp(0)
+            repo_type_q = Q(pulp2_repo_type=plugin_plan.type)
+            imp_type_q = Q(pulp2_type_id__in=importer_types)
+            dist_type_q = Q(pulp2_type_id__in=distributor_types)
+
+            plugin_pulp2repos = Pulp2Repository.objects.filter(repo_type_q)
+            repo_premigrated_last_by_added = plugin_pulp2repos.aggregate(
+                Max('pulp2_last_unit_added')
+            )['pulp2_last_unit_added__max'] or epoch
+            repo_premigrated_last_by_removed = plugin_pulp2repos.aggregate(
+                Max('pulp2_last_unit_removed')
+            )['pulp2_last_unit_removed__max'] or epoch
+            imp_premigrated_last = Pulp2Importer.objects.filter(imp_type_q).aggregate(
+                Max('pulp2_last_updated')
+            )['pulp2_last_updated__max'] or epoch
+            dist_premigrated_last = Pulp2Distributor.objects.filter(dist_type_q).aggregate(
+                Max('pulp2_last_updated')
+            )['pulp2_last_updated__max'] or epoch
+
+            is_content_added_q = mongo_Q(last_unit_added__gte=repo_premigrated_last_by_added)
+            is_content_removed_q = mongo_Q(last_unit_removed__gte=repo_premigrated_last_by_removed)
+            is_new_enough_repo_q = is_content_added_q | is_content_removed_q
+            is_empty_repo_q = mongo_Q(last_unit_added__exists=False)
+            is_new_enough_imp_q = mongo_Q(last_updated__gte=imp_premigrated_last)
+            is_new_enough_dist_q = mongo_Q(last_updated__gte=dist_premigrated_last)
+            repo_repo_id_q = mongo_Q(repo_id__in=repos)
+            imp_repo_id_q = mongo_Q(repo_id__in=importers_repos)
+            dist_repo_id_q = mongo_Q(repo_id__in=distributors_repos)
+
+            updated_importers = Importer.objects(
+                imp_repo_id_q & is_new_enough_imp_q
+            ).only('repo_id')
+            updated_imp_repos = set(imp.repo_id for imp in updated_importers)
+            updated_distributors = Distributor.objects(
+                dist_repo_id_q & is_new_enough_dist_q
+            ).only('repo_id')
+            updated_dist_repos = set(dist.repo_id for dist in updated_distributors)
+            updated_impdist_repos = updated_imp_repos | updated_dist_repos
+
+            mongo_updated_repo_q = repo_repo_id_q & (is_new_enough_repo_q | is_empty_repo_q)
+            mongo_updated_imp_dist_repo_q = mongo_Q(repo_id__in=updated_impdist_repos)
+
+            mongo_repo_qs = Repository.objects(
+                mongo_updated_repo_q | mongo_updated_imp_dist_repo_q
+            ).order_by('last_unit_added')
+
+            pb.total += mongo_repo_qs.count()
```

Comment:
In the code below I would recommend putting the ```pb.increment()``` inside the transaction.

Also it doesn't look like ```notes``` is used by the function below, or by any of the ```pre_migrate_*``` functions, maybe we can drop it from the query.